### PR TITLE
[Feature][DataSource]Add flink datasource plugin

### LINKED ISSUE #15630: [Feature][Datasource] Add Flink Datasource
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Currently spark use thrift server for replacement of hiveserver2 to provide Spark SQL capability.
https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/sql-gateway/overview/
Flink SQL Gateway is also a replacement solution of hiveserver2.

### Use case

https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/sql-gateway/overview/

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by pegasas**
## Purpose of the pull request

fix: https://github.com/apache/dolphinscheduler/issues/15630

[Feature][Datasource] add flink datasource

## Brief change log

Currently spark use thrift server for replacement of hiveserver2 to provide Spark SQL capability.
https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/sql-gateway/overview/
Flink SQL Gateway is also a replacement solution of hiveserver2.

## Verify this pull request

image
image
image
image

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Review by songjianet — CHANGES_REQUESTED** [review]
image

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-flink/src/main/java/org/apache/dolphinscheduler/plugin/datasource/flink/param/FlinkDataSourceProcessor.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+        String[] tmpArray = kyuubiConnectionParam.getAddress().split(Constants.DOUBLE_SLASH);
+        StringBuilder hosts = new StringBuilder();
+        String[] hostPortArray = tmpArray[tmpArray.length - 1].split(Constants.COMMA);
+        for (String hostPort : hostPortArray) {
+            hosts.append(hostPort.split(Constants.COLON)[0]).append(Constants.COMMA);
+        }
+        hosts.deleteCharAt(hosts.length() - 1);
+        flinkDataSourceParamDTO.setHost(hosts.toString());
+        flinkDataSourceParamDTO.setPort(Integer.parseInt(hostPortArray[0].split(Constants.COLON)[1]));
```
[the remark this answers is not in the document]
Comment (pegasas) [reply dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-flink/src/main/java/org/apache/dolphinscheduler/plugin/datasource/flink/param/FlinkDataSourceProcessor.java:70]: Ack.
Thanks for calling out.

**Comment by pegasas** [comment]
Caused by: java.sql.SQLFeatureNotSupportedException: FlinkConnection#prepareStatement is not supported yet.
	at org.apache.flink.table.jdbc.BaseConnection.prepareStatement(BaseConnection.java:42)
	at org.apache.flink.table.jdbc.FlinkConnection.prepareStatement(FlinkConnection.java:49)
	at org.apache.dolphinscheduler.plugin.task.sql.SqlTask.prepareStatementAndBind(SqlTask.java:380)
	... 8 common frames omitted

will do some compacity work on sql task

**Comment by pegasas** [comment]
> This pull request has been automatically marked as stale because it has not had recent activity for 120 days. It will be closed in 7 days if no further activity occurs.

remove stale

**Pull request closed without merging**